src/transports/dryModeRunner.ts

```diff
@@ -0,0 +1,79 @@
+import { InMemoryTransport } from "./inMemoryTransport.js";
+import { TransportRunnerBase, type TransportRunnerConfig } from "./base.js";
+import { type Server } from "../server.js";
+
+export type DryModeTestHelpers = {
+    exit(this: void, exitCode: number): never;
+    logger: {
+        log(this: void, message: string): void;
+        error(this: void, message: string): void;
+    };
+};
+
+type DryModeRunnerConfig = TransportRunnerConfig & DryModeTestHelpers;
+
+const defaultLogger: DryModeTestHelpers["logger"] = {
+    log(message) {
+        console.warn(message);
+    },
+    error(message) {
+        console.error(message);
+    },
+};
+
+export class DryModeRunner extends TransportRunnerBase {
+    private server: Server | undefined;
+    private exitProcess: DryModeTestHelpers["exit"];
+    private consoleLogger: DryModeTestHelpers["logger"];
```

Comment:
```suggestion
    private logger: DryModeTestHelpers["logger"];
```
(nit) this also isn't dependent on a console